genie-web/src/main/java/com/netflix/genie/web/health/GenieCpuHealthIndicator.java

```diff
@@ -38,36 +41,58 @@
 public class GenieCpuHealthIndicator implements HealthIndicator {
     private static final String CPU_LOAD = "cpuLoad";
     private final double maxCpuLoadPercent;
-    private final OperatingSystemMXBean operatingSystemMXBean;
+    private final int maxCpuLoadConsecutiveOccurrences;
+    private final DistributionSummary summaryCpuMetric;
+    private int cpuLoadConsecutiveOccurrences;
     /**
      * Constructor.
      *
      * @param healthProperties The maximum physical memory threshold
+     * @param registry         Registry
+     * @param taskScheduler    task scheduler
      */
     @Autowired
     public GenieCpuHealthIndicator(
-        @NotNull final HealthProperties healthProperties
-        ) {
+        @NotNull final HealthProperties healthProperties,
+        @NotNull final Registry registry,
+        @NotNull final TaskScheduler taskScheduler
+    ) {
         this(healthProperties.getMaxCpuLoadPercent(),
-            (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean());
+            healthProperties.getMaxCpuLoadConsecutiveOccurrences(),
+            (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean(),
+            registry.distributionSummary("genie.cpuLoad"),
+            taskScheduler);
     }
 
     GenieCpuHealthIndicator(final double maxCpuLoadPercent,
-                            final OperatingSystemMXBean operatingSystemMXBean) {
+        final int maxCpuLoadConsecutiveOccurrences,
+        final OperatingSystemMXBean operatingSystemMXBean,
+        final DistributionSummary summaryCpuMetric,
+        final TaskScheduler taskScheduler) {
         this.maxCpuLoadPercent = maxCpuLoadPercent;
-        this.operatingSystemMXBean = operatingSystemMXBean;
+        this.maxCpuLoadConsecutiveOccurrences = maxCpuLoadConsecutiveOccurrences;
+        this.summaryCpuMetric = summaryCpuMetric;
+        this.summaryCpuMetric.record((long) (operatingSystemMXBean.getSystemCpuLoad() * 100));
+        taskScheduler.scheduleAtFixedRate(() -> this.summaryCpuMetric
+            .record((long) (operatingSystemMXBean.getSystemCpuLoad() * 100)), 5000);
     }
 
     @Override
     public Health health() {
-        final double currentCpuLoadPercent = operatingSystemMXBean.getSystemCpuLoad() * 100;
+        final long cpuCount = summaryCpuMetric.count();
+        final long cpuTotal = summaryCpuMetric.totalAmount();
+        final double currentCpuLoadPercent = cpuCount == 0 ? 0 : (cpuTotal / (double) cpuCount);
```

Comment:
This will not be the current cpu load, but rather the average since the class was instantiated. `totalAmount` and `count` return cumulative values since the meter was created.